devtools/x/src/lint/guppy.rs

```diff
@@ -507,3 +507,167 @@ impl PackageLinter for OnlyPublishToCratesIo {
         Ok(RunStatus::Executed)
     }
 }
+
+// Ensure that Move crates do not depend on Diem crates.
+#[derive(Debug)]
+pub struct MoveCratesDontDependOnDiemCrates;
+
+// List of existing move to diem dependencies.
+// The goal is eliminate all of them gradually.
+// In general, you should refrain from adding new entries to this list.
+const EXISTING_MOVE_TO_DIEM_DEPS: &[(&str, &str)] = &[
+    ("language-benchmarks", "language-e2e-tests"),
+    ("language-benchmarks", "diem-vm"),
+    ("language-benchmarks", "diem-types"),
+    ("language-benchmarks", "diem-state-view"),
+    ("language-benchmarks", "diem-proptest-helpers"),
+    ("bytecode-verifier", "diem-types"),
+    ("bytecode-verifier-tests", "diem-types"),
+    ("invalid-mutations", "diem-types"),
+    ("invalid-mutations", "diem-proptest-helpers"),
+    ("compiler", "diem-types"),
+    ("bytecode-source-map", "diem-types"),
+    ("ir-to-bytecode", "diem-types"),
+    ("ir-to-bytecode-syntax", "diem-types"),
+    ("compiled-stdlib", "diem-types"),
+    ("compiled-stdlib", "diem-proptest-helpers"),
+    ("compiled-stdlib", "diem-framework"),
+    ("compiled-stdlib", "diem-crypto"),
+    ("move-ir-types", "diem-types"),
+    ("move-lang", "fallible"),
+    ("move-lang", "diem-types"),
+    ("move-lang", "datatest-stable"),
+    ("move-lang-functional-tests", "functional-tests"),
+    ("move-lang-functional-tests", "diem-types"),
+    ("move-lang-functional-tests", "datatest-stable"),
+    ("move-lang-test-utils", "datatest-stable"),
```

Comment:
I think datatest-stable isn't really a "deim" dependency. Its only used by move tests IIRC

Reply:
diem-framework also uses it.

Reply:
So here what matters to us is not whether a crate is conceptually Diem but whether it resides in the Diem repo. I don't see us brining datatest-stable to the new repo so this is also a dependency we have to remove. 

(BTW what was the original reason we forked datatest in the first place? Did it have something to do with the TCB? cc @sunshowers)

Reply:
We reimplemented datatest to make it work on stable rust.

Reply:
datatest-stable can move into the diem-devtools repo

Reply:
In this case, since it probably does not need updating, we could just have it internal to both places. Or publish it seprately on crates.io and put it in its own repo.